### PR TITLE
Links within links

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

`<a>` tags can't be embedded inside `<a>` tags. It's a HTML thing. But we need this functionality. For instance Facia cards must be clickable, but sublinks inside the cards must also be clickable, and they must redirect to a different location.

We have been working around this by embedding clickable overlays and tweaking their z-indexes, adding classes in JavaScript to apply styling such as darkening of the background colour. This is a bit flaky (hovering over sublinks causes background colour to lighten again) and involves a lot of extra CSS and HTML.

This change adds `<object>` tags to surround `<a>` tags that would appear inside other `<a>` tags. This is kinda legit HTML, as browsers don't need to validate the content of `<object>` tags. 

It works in all browsers down to (and including) IE9.

There are a lot of changes here, I can appreciate this is hard to review and will require a fair bit of testing. I'm happy to reduce the scope of this PR to just Facia, Rich Links and Related Links if this would make it easier.

## What is the value of this and can you measure success?

Removes extraneous HTML and CSS. Makes Garnett simpler.

## Does this affect other platforms - Amp, Apps, etc?

No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

Some styles are affected, but there should be no functional changes.

## Tested in CODE?

- [x] Tested in CODE

### CODE REVIEW

**Comment by NataliaLKB** [comment]
I think this is actually not a bad idea! I would want some way of noting in the code why we are using `<object>` though so it doesn't get deleted down the road. Certainly its not obvious in some of the cases.

**Comment by paperboyo** [comment]
Wooo, coool! May require adjusting Mr. Gustaf. Specifically, around [this respectful comment](https://github.com/guardian/gustaf/blob/23f0905f58a70767b56a17e102d0c1911e28406e/src/images.js#L32).  😄

**Comment by SiAdcock** [comment]
Good point, @NataliaLKB , I've added comments in the relevant places. Also given the object type a custom meaningful name.

@paperboyo Thanks for the tip! I'll take a look at gustaf and see what needs doing

**Comment by paperboyo** [comment]
@SiAdcock Thank you! (hopefully: nothing :) )

**Comment by SiAdcock** [comment]
@paperboyo very little!

guardian/gustaf#61

**Comment by sndrs** [comment]
amazing!

**Review by sndrs — APPROVED** [review]
lgtm! nice one!

File: static/src/stylesheets/base/_faux-block-link.scss [hunk 1]
```diff
@@ -1,67 +0,0 @@
-/*doc
```
Comment (NataliaLKB) [inline static/src/stylesheets/base/_faux-block-link.scss]: ❤️

**Review by NataliaLKB — APPROVED** [review]
This seems much better to me! Awesome job :1st_place_medal:

**Comment by SiAdcock** [comment]
Problems with this PR

## Layout thrashing on fronts

dec-12-2017 15-46-47

This is something to do with the way the `<object>` fallback is rendered. It looks like some browser logic is taking ~0.5s to decide whether to render the fallback. This can be mitigated by setting the height and width of the `<object>` element to `0` in CSS. There may be other solutions.

## Repeated headline in onward containers**

pasted image at 2017_12_12 04_45 pm

These containers are rendered as anchors, but it appears there's a nested anchor inside them. This needs to be wrapped in an `<object>`  element. I've corrected several similar issues and missed this one. For some reason, this is how nested anchors manifests on the painted page.

## Blue headlines

All cards on the front have blue headlines. In most cases, only the kicker should be blue, headlines should be black.

## Rich link styling

I noticed some odd styling in rich links. On hover, `text-decoration: underline` was erroneously being applied to the headline.

**Pull request merged**